### DISCUSSION TITLE
.direct_hemi method

### DISCUSSION THREAD

**Discussion by robertweitl** [LBNL-ETA/pyradiance#43] [Q&A] [answered]
Dear community, Dear Taoning,
for our mutual learning:

What is the result of this direct-hemi method? Or put it another way: why does this value differ from the ladybug-bsdf-viewer value for 
Outgoing Hemisphere (visible transmission front)
Direct Hemispherical: 90.10%

From the method I get:
bsdf_temp.direct_hemi(0, 0, 't'))  # direct hemispherical transmittance at theta=0°, phi=0
0.8977977210016154

I would have expected these to values to be the same? Is the method not only selecting the patch of the incident direction?

the file: https://drive.google.com/file/d/19Iy9qtaC-HVGie-ENnTaA9Eet4_-DMmB/view?usp=sharing

**Comment by taoning** [comment, marked as answer]
The two values should be the same.  My guess would be the mismatch between floating point parsing and multiplication between the two languages, and possibly the solid angle per patch.
